In [1]:
from scipy import stats
import numpy as np

## Задача 1
Из партии батареек с номинальным напряжением 12 В наугад выбрано семь штук. Согласно показаниям высокоточного вольтметра, фактически выдаваемое ими напряжение составило: 12,9; 11,6; 13,5; 13,9; 12,1; 11,9; 13,0 В соответственно. Можно ли считать, что среднее напряжение действительно равно 12 В?

In [2]:
voltage = np.array([12.9, 11.6, 13.5, 13.9, 12.1, 11.9, 13.0])

In [3]:
voltage.mean()

12.700000000000001

In [4]:
voltage.std()

0.7946247991527647

In [5]:
stats.ttest_1samp(voltage, 12, alternative='two-sided')

Ttest_1sampResult(statistic=2.1578017974978794, pvalue=0.07430190210562303)

Не можем отвергнуть нулевую гипотезу, т к p-value больше 0.05 => мы можем действительно утверждать, что среднее напряжение действительно равно 12 и среднее выборочное 12.7 получилось случайно


## Задача 2

Есть два различных метода определения процентного содержания жира в мясе. Доля жира для 10 образцов мяса была определена каждым из двух методов. Позволяют ли полученные результаты сделать вывод о том, что один метод даёт в среднем более высокие показания, чем другой?

In [6]:
method1 = np.array([23.1, 23.2, 26.5, 26.6, 27.1, 48.3, 40.5, 25.0, 38.4, 23.5])
method2 = np.array([22.7, 23.6, 27.1, 27.4, 27.4, 46.8, 40.4, 24.9, 38.1, 23.8])

In [7]:
diff = method1 - method2

In [8]:
diff

array([ 0.4, -0.4, -0.6, -0.8, -0.3,  1.5,  0.1,  0.1,  0.3, -0.3])

Ho - средняя разница равна 0

Ha - средняя разница не равна 0

In [9]:
stats.ttest_1samp(diff, 0, alternative='two-sided')

Ttest_1sampResult(statistic=1.7154884978678027e-15, pvalue=0.9999999999999987)

p-value больше 0.05 - значит оба метода дают примерно одинаковые результаты

## Задача 3
Для проверки гипотезы о том, что жители городов имеют в среднем более высокое систолическое артериальное давление, чем жители сельской местности, было проведено обследование 10 горожан и 9 жителей села примерно одного возраста. Получены следующие данные об уровне давления для горожан: 132, 111, 119, 138, 200, 131, 138, 170, 159, 140; для селян: 115, 190, 127, 155, 148, 121, 116, 121, 197. Можно ли считать верной выдвинутую гипотезу на уровне значимости 0,05?

In [10]:
citizens = [132, 111, 119, 138, 200, 131, 138, 170, 159, 140]
villagers = [115, 190, 127, 155, 148, 121, 116, 121, 197]

In [11]:
stats.ttest_ind(citizens, villagers, alternative='greater')

Ttest_indResult(statistic=0.035165199049777565, pvalue=0.486178812200461)

p-value больше 0.05 - мы не можем сказать, что у горожан давление больше

## Задача 4
Известно, что 52% новорождённых детей — мальчики, а 48% — девочки. В случайной выборке из 5 000 человек в возрасте от 30 до 60 лет мужчин оказалось 2 500. Можно ли на уровне значимости 0,05 считать, основываясь на этих данных, что смертность среди мужчин и женщин одинакова?

Сведем задачу к тому, что доля новорожденных мальчиков и доля мужчин в возрасте от 30 до 60 лет не отличается.

Тогда Ho - доля новорожденных мальчиков равна доле мужчин во взрослом возрасте (от 30 до 60 лет), Ha - доля новорожденных мальчиков не равна доле мужчин во взрослом возрасте (от 30 до 60 лет)

In [19]:
n = 5000
m = 2600
p0 = 0.5

In [20]:
T = (m - n*p0)/((n*p0*(1 - p0))**0.5)
P = 1 - stats.norm.cdf(T)
print("Statistic: ",T,", p-value: ",P)

Statistic:  2.82842712474619 , p-value:  0.0023388674905235884


Можем отвергнуть нулевую гипотезу, что смертность мужчин и женщин одинакова, т к p-value меньше 0.05

## Задача 5
В первом эксперименте для сравнения двух средних X1=17, X2=16 применялся t-критерий Стьюдента, и эти различия оказались значимы (p = 0, 001).

Во втором исследовании, также при помощи t - критерия, сравнивались два средних X1=17, X2=36, и эти различия не значимы при p = 0,8. В чем может быть причина таких результатов, при условии, что нулевая и альтернативная гипотезы в обоих случаях одинаковы?

Причина может заключаться в разном количестве наблюдений и в разной дисперсии.